app/app.go

```diff
@@ -0,0 +1,199 @@
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"strings"
+
+	"github.com/datatrails/go-datatrails-logverification/logverification/app"
+)
+
+/**
+ * Merklelog related APP contents
+ *
+ * An APP in the context of merklelog is an interface that commits log entries.
+ *
+ * Apps include:
+ *  - assetsv2
+ *  - eventsv1
+ */
+
+const (
+	AssetsV2AppDomain = byte(0)
+	EventsV1AppDomain = byte(1)
+)
+
+var (
+	ErrNoJsonGiven = errors.New("no json given")
+)
+
+// AppDataToVerifiableLogEntries converts the app data to verifiable log entries
+func AppDataToVerifiableLogEntries(appData []byte, logTenant string) ([]app.AppEntry, error) {
+
+	// first attempt to convert the appdata to a list of events
+	eventList, err := eventListFromData(appData)
+	if err != nil {
+		return nil, err
+	}
+
+	// now we have an event list we can decipher if the app is
```

Comment:
Is it not possible that its a mixture of both?

Reply:
Urmmm yes it is, technically if a user decided to transform our API responses, they could have a mixture of both assetsv2 and eventsv1 events.

However this is probably unlikely?

Its a bit inconvenient at the moment, but you could just use two lists from the separate API's

Reply:
It might be worth us surfacing this information in a comment, but its not a huge thing - if they're reading the code, I suppose they'll see what its doing

Reply:
Are you happy to call it TD.

It would be great to have a list of mixed events, I don't think customers will get too mixed up by it as we have separate APIs.

However in the future as we add more app domains it could become a sticking point?